lib/expression_parser.rb

```diff
@@ -18,6 +18,13 @@ def write_vm_code(vm_writer)
   end
 end
 
+Boolean = Struct.new(:value) do
+  def write_vm_code(vm_writer)
+    vm_writer.write_push(:CONST, 0)
+    vm_writer.write_arithmetic(:NOT) if value == :TRUE
```

Comment:
Yeah, I guess this is neater than having to conditionally `push constant 1` and then `neg`.

Reply:
Yeah my first instinct wasn't the best solution. Using `push constant 0` then `not` makes it extra clear that it's a `true`.